sorting/selection_sort_recursive.c

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file
+ * @author [Dhruv Pasricha](https://github.com/DhruvPasricha)
+ * @brief [Selection Sort](https://en.wikipedia.org/wiki/Selection_sort)
+ * implementation using recursion.
+ */
+ 
+#include <assert.h>     /// for assert
+#include <stdio.h>      /// for IO operations
+#include <stdlib.h>     /// for dynamic memory allocation
+#include <time.h>       /// for random numbers generation
+#include <inttypes.h>   /// for uint8_t
+
+/**
+ * @brief Swapped two numbers using pointer
+ * @param first pointer of first number
+ * @param second pointer of second number
+ */
+void swap(uint8_t *first, uint8_t *second)
+{
+    uint8_t temp = *first;
+    *first = *second;
+    *second = temp;
+}
+
+/**
+ * @brief Returns the index having minimum value using recursion
+ * @param arr array to be sorted
+ * @param size size of array
+ * @return min_index index of an element having a minimum value
+*/
+uint8_t findIndex(const uint8_t *arr, const uint8_t size)
+{
+    if (size == 1)
+    {
+        return 0;
+    }
+
+    // marking recursive call to reach starting element
+    uint8_t min_index = findIndex(arr, size - 1);
+
+    if (arr[size - 1] < arr[min_index])
+    {
+        min_index = size - 1;
+    }
+
+    return min_index;
+}
+
+/**
+ * @brief Selection Sort algorithm implemented using recursion
+ * @param arr array to be sorted
+ * @param size size of the array
+ * @returns void
+ */
+void selectionSort(uint8_t *arr, const uint8_t size)
```

Comment:
here also consider the same suggestion.

Reply:
I have made the required changes, please see to it.